client/components/editor/teaching-elements/Assessment/TextSelection/helpers.js

```diff
@@ -0,0 +1,101 @@
+import Blast from 'blast-vanilla';
+import first from 'lodash/first';
+import generateRange from 'lodash/range';
+import isEmpty from 'lodash/isEmpty';
+import last from 'lodash/last';
+
+const blastClass = 'blast';
+const wordClass = 'word';
+const blastWordSelector = `span.${blastClass}.${wordClass}`;
+const idPrefix = 'text-';
+
+const isText = node => node.nodeType === Node.TEXT_NODE;
+
+export const isSelected = node => node.hasAttribute('range');
+
+export function extractIndex(node) {
+  const index = node.id.substr(idPrefix.length);
+  return parseInt(index, 10);
+}
+
+export function generateStructure(content, mapper) {
+  if (!content) return '';
+  const container = document.createElement('div');
+  container.innerHTML = content;
+  // eslint-disable-next-line no-new
+  new Blast(container);
+  const iterator = document.createNodeIterator(container, NodeFilter.SHOW_TEXT);
+  const nodes = [];
+  while (iterator.nextNode()) { nodes.push(iterator.referenceNode); }
```

Comment:
Previous three lines are completely generic and can be extracted to `filterNodes(nodeFilter)` function :bowtie: